lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/navigator/Navigator.java

```diff
@@ -266,10 +266,20 @@ CoordinatorLayout getOverlaysLayout() {
     }
 
     public void onHostPause() {
-        super.onViewDisappear();
+        overlayManager.onHostPause();
+        if (!modalStack.isEmpty()) {
```

Comment:
I think this is not 100% correct. If the currently presented modal is displayed with `modalPresentationStyle: 'overCurrentContext'`, then the root layout also needs to get a disappear event.

Reply:
hmm, agree, knew that I'm missing something, checking options will do the work.
Thx dude